src/main/java/frc/robot/subsystems/Arm.java

```diff
@@ -7,90 +7,124 @@
+      new RobotPreferences.BooleanValue("Arm", "Enable Tab", false);
 
+  private final String name;
   private final TalonFX motor;
   private final DutyCycleEncoder absoluteEncoder;
-
-  private final TrapezoidProfile profile;
-
-  private final ArmFeedforward feedForward;
-  private final ProfiledPIDController controller;
-
-  private final TrapezoidProfile.State currentAngle = new TrapezoidProfile.State();
-  private final TrapezoidProfile.State goalAngle = new TrapezoidProfile.State();
+  private double currentAngle = 0;
+  private double goalAngle = 0;
   private boolean enabled;
-  private final Timer timer = new Timer();
-  private double currentAngleTime = 0;
+  private final double rotationsPerRadians;
 
   /** Creates a new Arm. */
   public Arm(ArmParameters parameters) {
+    name = parameters.toString();
+    rotationsPerRadians = parameters.getGearRatio() / (2 * Math.PI);
 
-    feedForward = parameters.getArmFeedforward();
-    profile = new TrapezoidProfile(parameters.getConstraints());
-    controller = parameters.getProfiledPIDController();
+    TalonFXConfiguration talonFXConfigs = new TalonFXConfiguration();
 
     motor = new TalonFX(parameters.getMotorID());
     absoluteEncoder = new DutyCycleEncoder(parameters.getEncoderID());
 
-    MotorOutputConfigs configs = new MotorOutputConfigs();
-    configs.NeutralMode = NeutralModeValue.Brake;
-    configs.Inverted = InvertedValue.Clockwise_Positive;
-    motor.getConfigurator().apply(configs);
+    MotorOutputConfigs motorOutputConfigs = talonFXConfigs.MotorOutput;
+    motorOutputConfigs.NeutralMode = NeutralModeValue.Brake;
+    motorOutputConfigs.Inverted = InvertedValue.Clockwise_Positive;
+    // set slot 0 gains
+    Slot0Configs slot0Configs = talonFXConfigs.Slot0;
+    slot0Configs.kS = parameters.getkS();
+    /** Need to convert kV and kA from radians to rotations. */
+    slot0Configs.kV = parameters.getkV() * 2 * Math.PI;
+    slot0Configs.kA = parameters.getkA() * 2 * Math.PI;
+    slot0Configs.kP = 0;
+    slot0Configs.kI = 0;
+    slot0Configs.kD = 0;
+
+    // set Motion Magic Expo settings
+    MotionMagicConfigs motionMagicConfigs = talonFXConfigs.MotionMagic;
+    motionMagicConfigs.MotionMagicCruiseVelocity = 0;
+    motionMagicConfigs.MotionMagicExpo_kV = slot0Configs.kV;
+    motionMagicConfigs.MotionMagicExpo_kA = slot0Configs.kA;
+
+    motor.getConfigurator().apply(talonFXConfigs);
+
     absoluteEncoder.setDutyCycleRange(1.0 / 1025.0, 1024.0 / 1025.0);
   }
 
   /** Updates the sensor state. */
   private void updateSensorState() {
-    double newAngle = absoluteEncoder.get() * (2 * Math.PI);
-    double currentTime = Timer.getFPGATimestamp();
-    if (currentAngleTime != 0) {
-      currentAngle.velocity = (newAngle - currentAngle.position) / (currentTime - currentAngleTime);
-    }
-    currentAngle.position = newAngle;
-    currentAngleTime = currentTime;
+    currentAngle = absoluteEncoder.get() * (2 * Math.PI);
```

Comment:
Since we're relying on the motor encoder in Motion Magic, we should get the current angle from it rather than the absolute encoder. We'll want to display both the motor encoder and absolute encoder readings on the `Status` layout of Shuffleboard. We will also likely need and offset between the absolute and motor encoder readings. For now, it can be 0.